tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs

```diff
@@ -46,9 +47,10 @@ internal ILocalRepositoryProvider GetLocalProvider(string rootPath)
         var providerPtr = IntPtr.Zero;
         try
         {
-            // "BDA76685-E749-4f09-8F13-C466D0802DA1" is the hardcoded GUID for the git extension. In a future change, the value will no longer be hard-coded and
-            // will be discoverable given a root path from a storage location
-            var hr = PInvoke.CoCreateInstance(Guid.Parse("BDA76685-E749-4f09-8F13-C466D0802DA1"), null, CLSCTX.CLSCTX_LOCAL_SERVER, typeof(ILocalRepositoryProvider).GUID, out var extensionObj);
+            var repositoryTracker = new RepositoryTracking(null);

```

Comment:
This function will be called frequently and needs to return as quickly as possible. But RepositoryTracking ctor reloads the list of repositories from disk. Instead, cache this object and have it reload only when a repo is added/removed/changed.

Reply:
Resolved in latest. Thanks!